libethereum/EthereumHost.h

```diff
@@ -51,20 +51,95 @@ class TransactionQueue;
 class BlockQueue;
 class BlockChainSync;
 
+struct EthereumPeerStatus
+{
+    /// What, if anything, we last asked the other peer for.
+    Asking m_asking = Asking::Nothing;
+    /// When we asked for it. Allows a time out.
+    std::atomic<time_t> m_lastAsk;
+    /// Peer's protocol version.
+    unsigned m_protocolVersion;
+    /// Peer's network id.
+    u256 m_networkId;
+    /// These are determined through either a Status message or from NewBlock.
+    /// Peer's latest block's hash that we know about or default null value if no need to sync.
+    h256 m_latestHash;
+    /// Peer's latest block's total difficulty.
+    u256 m_totalDifficulty;
+    h256 m_genesisHash;            ///< Peer's genesis hash
+    /// Protocol version this peer supports received as capability
+    u256 m_peerCapabilityVersion;
```

Comment:
Probably can be smaller type.

Reply:
In general it comes from devp2p handshake protocol where capability version can be any positive integer, so probably I'd keep it as `u256` in libp2p interface, but I guess for eth implementation we can narrow it here.

Reply:
Actually there's no much sense it saving it (it is always 63 now), so I removed it.